### PR TITLE
Add support for CompletionList "applyKind" to control how defaults and per-item commitCharacters/data are combined

### PR DESCRIPTION
**Pull request by DanTup**
Implements the changes in the LSP spec PR at https://github.com/microsoft/language-server-protocol/pull/2018 (see also https://github.com/microsoft/language-server-protocol/issues/1802).

Please let me know if you think there are any additional test cases that need covering.

### CODE REVIEW

**Comment by DanTup** [comment]
@dbaeumer FYI in case you didn't see this :)

Looks like one of the status checks is stuck too (not the approval one, but `vscode-languageserver-node`).

**Comment by dbaeumer** [comment]
I was out the last 1 1/2 weeks. Will have a look this week.

File: types/src/main.ts [hunk 1]
```diff
@@ -2220,6 +2220,36 @@ export namespace InsertTextMode {
 
 export type InsertTextMode = 1 | 2;
 
+/**
+ * Defines how values from a set of defaults and an individual item will be
+ * merged.
+ *
+ * @since 3.18.0
+ */
+export namespace ApplyKind {
```
[the remark this answers is not in the document]
Comment (DanTup) [reply types/src/main.ts:2229]: Sure - I've pushed f8ed433e951cd5da164abd0d9fbe1d275b2514a0 + 5e2874589df417071865002e7da6ffd507da7cb3 with this change and some minor tweaks, and equiv changes in https://github.com/microsoft/language-server-protocol/pull/2018

**Review by dbaeumer — APPROVED** [review]

**Review by joyceerhl — APPROVED** [review]

**Comment by DanTup** [comment]
@dbaeumer thanks! Are you planning a new `-next` build in the near future? (not a request for you to do one, it'll just help me plan for when to start implementing this for real in the Dart server). Thanks! :)

**Comment by dbaeumer** [comment]
@DanTup yes, will do one beginning of endgame week.

**Pull request merged**